### PR TITLE
Feature 7: Pagination 

### PR DESCRIPTION
**Pull request by saa-wtag**
[[Bonus] Task 8: Pagination](https://trello.com/c/B8XnBy4w)
This PR introduces the functionality of pagination the todos by clicking the Load more button at bottom.

**Functionalities:**
- By clicking load more button next tasks will be shown on the page upto next 9 or less (if next tasks no is less than 9) tasks.

Please have a close look on this PR and kindly put your valuable comments.

https://github.com/saa-wtag/ToDoApp_VanillaJS/assets/164306406/b339aa5d-b436-4edd-96eb-f7850fa4d464

### CODE REVIEW

**Review by sal-wtag — APPROVED** [review]

**Review by maa-wtag — COMMENTED** [review]
please resolve the PR conflicts first

**Comment by saa-wtag** [comment]
> please resolve the PR conflicts first

Done, bhai.

File: styles.css [hunk 1]
```diff
@@ -249,3 +249,14 @@ body {
...
 }
+
+.load-more-button {
+  background-color: var(--secondary-color);
+  color: var(--white);
+  border-radius: 4px;
+  padding: 0.4rem 1rem;
+  display: block;
+  margin: 0 auto;
+  margin-top: 0.4%;
```
Comment (maa-wtag) [inline styles.css]: could we use a proper value instead, using % in CSS box model elements is not a good practice.

File: js/script.js [hunk 2]
```diff
@@ -19,12 +20,13 @@ import {
...
-import { MESSAGES } from "./const.js";
+import { MESSAGES, TASK_PER_PAGE } from "./const.js";
 
 let tasks = [];
 let isTaskInputVisible = false;
 let currentFilter = "all";
 let filteredOrSearchableTasks = [];
+let tasksDisplayed = TASK_PER_PAGE;
 
 const filteredSearchableTasksOrTasks = filteredOrSearchableTasks.length
```
Comment (maa-wtag) [inline js/script.js]:
```suggestion
const finalTaskList = filteredOrSearchableTasks.length
```

File: js/script.js [hunk 3]
```diff
@@ -124,19 +127,31 @@ const filterTasks = (searchTitle = "") => {
...
 };
 
+$loadMore.addEventListener("click", () => {
+  tasksDisplayed = Math.min(
+    tasksDisplayed + TASK_PER_PAGE,
+    filteredOrSearchableTasks.length || tasks.length
+  );
+  renderTasks(filteredSearchableTasksOrTasks);
+});
+
```
Comment (maa-wtag) [inline js/script.js]:
```suggestion
$loadMore.addEventListener("click", () => {
  if (!Array.isArray(tasks) || !Array.isArray(filteredOrSearchableTasks)) {
    console.error("Task arrays are not defined correctly.");
    return;
  }

  tasksDisplayed = Math.min(
    tasksDisplayed + TASK_PER_PAGE,
    filteredOrSearchableTasks.length || tasks.length
  );

  if (typeof renderTasks !== 'function') {
    console.error("renderTasks is not a function");
    return;
  }

  renderTasks(finalTaskList);
});

```

**Review by maa-wtag — CHANGES_REQUESTED** [review]

File: js/script.js (hunk 3 above)
Comment (saa-wtag) [reply js/script.js]: Done, bhai.

File: js/script.js (hunk 2 above)
Comment (saa-wtag) [reply js/script.js]: Done, bhai.

File: styles.css (hunk 1 above)
Comment (saa-wtag) [reply styles.css]: Done, bhai

File: js/script.js [hunk 4]
```diff
@@ -124,19 +127,39 @@ const filterTasks = (searchTitle = "") => {
...
     containerBuilder(task, completeTask, editTask, deleteTask, updateTask);
   });
 
   $noTask.style.display = tasksToRender.length === 0 ? "flex" : "none";
+  $loadMore.style.display =
+    tasksToRender.length > tasksDisplayed ? "block" : "none";
 };
 
+$loadMore.addEventListener("click", () => {
+  if (!Array.isArray(tasks) || !Array.isArray(finalTaskList)) {
```
Comment (maa-wtag) [inline js/script.js]:
```suggestion
  if (!Array.isArray(tasks) || !Array.isArray(filteredOrSearchableTasks)) {
```

File: js/script.js [hunk 5]
```diff
@@ -124,19 +127,39 @@ const filterTasks = (searchTitle = "") => {
...
+    console.error("Task arrays are not defined correctly.");
+    return;
+  }
+  tasksDisplayed = Math.min(
+    tasksDisplayed + TASK_PER_PAGE,
+    finalTaskList.length || tasks.length
```
Comment (maa-wtag) [inline js/script.js]:
```suggestion
    filteredOrSearchableTasks.length || tasks.length
```

**Review by maa-wtag — CHANGES_REQUESTED** [review]

**Review by maa-wtag — APPROVED** [review]

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**